### PR TITLE
Fix/security groups update

### LINKED ISSUE #123: ssh_source_restriction is no longer available
The following change removed `ssh_source_restriction` https://github.com/cloudposse/terraform-aws-elastic-beanstalk-environment/commit/fe6d0d7242dff18928cc204fb3c693e104978b17#diff-7a370d8342e7203b805911c92454f0f4L494
but it is still present in the docs, and is also a variable that can be passed in, is this intentional?

I have checked a recent environment using the changes that have been made, and it no longer locks ssh down to the said source, but is now available to the world.

### PR DESCRIPTION
**Pull request by samcrudge**
## what

Suggested changed by @goruha. 

Default to empty string if aws_security_group.id is null. Empty string will then get filtered out by compact().
## why
Null value caused sort() function to crash, making it impossible to set create_security_group to false

Implementing changes suggested in  #229 by @goruha.
## references

Fixed sort function crashing when create_security_group=false #229 
Option create_security_group can't be false currently #216

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -644,7 +644,7 @@ resource "aws_elastic_beanstalk_environment" "default" {
   setting {
     namespace = "aws:autoscaling:launchconfiguration"
     name      = "SecurityGroups"
-    value     = join(",", compact(sort(concat([module.aws_security_group.id], var.associated_security_group_ids))))
+    value     = join(",", compact(sort(concat([module.aws_security_group.id == null ? "" : module.aws_security_group.id], var.associated_security_group_ids))))
```
Comment (joe-niland) [inline main.tf:647]: @samcrudge I had a look at this again and @goruha 's suggestion appears to be:

```suggestion
    value     = join(",", sort(compact(concat([module.aws_security_group.id], var.associated_security_group_ids))))
```

Did you try this one?

Comment (samcrudge) [reply main.tf:647]: Sorry i thought that was what i had committed! sorry!

Comment (joe-niland) [reply main.tf]: no worries! 
And so this branch is working with `create_security_group = false` ?

Comment (samcrudge) [reply main.tf]: Just doing some more testing on other projects we have blocked by this.

**Review by max-lobur — APPROVED** [review]

File: main.tf (hunk 1 above)
Comment (eanselmi) [reply main.tf]: I have my projects working perfect with the suggested changes, thanks @joe-niland it was very helpful!

Comment (eanselmi) [reply main.tf]: I'm using this line in my pipeline until this change is merged 

```
- name: Adjust beanstalk module PROD
      working-directory: ./environments/prod/.terraform/modules/beanstalk_production_environment
      run: sed -i 's|compact(sort(concat|sort(compact(concat|g' main.tf
```

Comment (joe-niland) [reply main.tf]: @samcrudge it's good to go from our side. Have you completed your testing?

Comment (samcrudge) [reply main.tf]: Yes I deployed to live environment and was successful

Comment (samcrudge) [reply main.tf]: I've tested across 2 projects successfully now. @joe-niland

**Comment by joe-niland** [comment]
Thanks for your contribution @samcrudge and @dawidrogowicz !

**Comment by dawidrogowicz** [comment]
cool, thanks for taking over :bow:

**Pull request merged**